frontend/gateway/forwarder/forward.go

```diff
@@ -161,20 +162,31 @@ func (c *bridgeClient) NewContainer(ctx context.Context, req client.NewContainer
 	}
 
 	for _, m := range req.Mounts {
-		var refProxy solver.ResultProxy
+		var workerRef *worker.WorkerRef
+		// TODO: What about ResultID?
+		// The *bridgeClient doesn't have a workerRefByID map.
```

Comment:
I am not sure this logic is valid, but you might be able to compare the ResultID against all refs in the `brdigeClient`, something like:
```
		if m.ResultID != "" {
			for _, ref := range c.refs {
				im, _ := ref.getImmutableRef(ctx)
				if m.ResultID == im.ID() {
					rr, _ := ref.ResultProxy.Result(ctx)
					workerRef, _ = rr.Sys().(*worker.WorkerRef)
					break
				}
			}
		}
```

Reply:
They don't share the same map unfortunately because with solve errors we are dealing with non-cached `solver.Result`.
I ended up resolving the same way I did in `llbBridgeForwarder`: https://github.com/moby/buildkit/pull/1732/commits/a799187d4f1e69758db97c6b7a0dae46057772b5